### PR TITLE
[RFC] Provide ParseCreateSession on conn to be used by both server and client

### PR DESCRIPTION
**Pull request by krsna1729**
Today we have duplicate code to fill up the session struct in examples for CSResponse. Can we use this method to populate the session struct for CSReq and Res

Signed-off-by: Saikrishna Edupuganti <saikrishna.edupuganti@intel.com>

### CODE REVIEW

**Comment by wmnsk** [comment]
It looks nice. Give me some more time to consider, too.
Thank you very much for your improving ideas and your patience 😉

**Comment by krsna1729** [comment]
Example code I'm using for CSReq handler

```
	csr := msg.(*messages.CreateSessionRequest)
	requiredTypesIEs := []struct {
		ieType uint8
		ie     *ies.IE
	}{
		{ies.IMSI, csr.IMSI},
		{ies.MSISDN, csr.MSISDN},
		{ies.MobileEquipmentIdentity, csr.MEI},
		{ies.APNRestriction, csr.APN},
		{ies.ServingNetwork, csr.ServingNetwork},
		{ies.RATType, csr.RATType},
		{ies.FullyQualifiedTEID, csr.SenderFTEIDC},
		{ies.BearerContext, csr.BearerContextsToBeCreated},
	}

	var requiredIEs []*ies.IE
	for _, req := range requiredTypesIEs {
		if req.ie == nil {
			// TODO: Respond with error
			return &v2.RequiredIEMissingError{Type: req.ieType}
		}
		requiredIEs = append(requiredIEs, req.ie)
	}

	session, err := c.ParseCreateSession(mmeAddr, requiredIEs...)
	if err != nil {
		return err
	}
	bearer := session.GetDefaultBearer()
```

File: v2/conn.go [hunk 1]
```diff
@@ -441,75 +441,59 @@ func (c *Conn) VersionNotSupportedIndication(raddr net.Addr, req messages.Messag
...
+// ParseCreateSession iterates through the ies and returns a session
+func (c *Conn) ParseCreateSession(raddr net.Addr, ie ...*ies.IE) (*Session, error) {
 	// retrieve values from IEs given.
 	sess := NewSession(raddr, &Subscriber{Location: &Location{}})
 	br := sess.GetDefaultBearer()
 	var err error
 	for _, i := range ie {
 		if i == nil {
-			continue
+			return nil, &RequiredIEMissingError{}
```
[the remark this answers is not in the document]
Comment (krsna1729) [reply v2/conn.go]: Oh OK. Will change that and resolve the conflict.

**Comment by krsna1729** [comment]
@wmnsk updated

**Pull request merged**